### PR TITLE
[documentation] #2193: Update README for macro crate

### PR DESCRIPTION
**Pull request by outoftardis**
Signed-off-by: Ekaterina Mekhnetsova <mekkatya@gmail.com>

### Description of the Change

Updated `macro` readme: https://github.com/outoftardis/iroha/tree/doc-macro/macro#iroha-macros

### Issue

Partially addresses #2193

### CODE REVIEW

File: macro/README.md [hunk 1]
```diff
@@ -1,66 +1,23 @@
...
+This crate contains macros and attributes for Iroha projects:
 
-This crate contains macros and attributes for Iroha projects.
-`iroha_derive` contains derive macros.
-
-### Features
-
-* `log` attribute for debugging functions inputs and output
+- `iroha_derive` is a custom derive macros for Iroha
+- `log` attribute is used for debugging function inputs and output
```
Comment (outoftardis) [inline macro/README.md:6]: I don't really see this `log` attribute in code so I'm not sure about this line

Comment (appetrosyan) [reply macro/README.md:6]: Remove it. We found that the `log` attribute causes deadlocks if not used carefully.

Comment (outoftardis) [reply macro/README.md]: removed

**Review by appetrosyan — APPROVED** [review]

File: macro/README.md [hunk 2]
```diff
@@ -1,66 +1,22 @@
...
+## Example
 
 ```rust
 #[derive(Clone, Debug, Encode, Decode)]
-pub struct Test [
-...
+pub struct Test {
+// ...
+}
 ```
```
Comment (Arjentix) [inline macro/README.md:22]: This example doesn't contain anything from this crate. I don't know why is it here originaly.
I'd suggest to replace it with example from [here](https://github.com/hyperledger/iroha/blob/61bbd74aec7c03da07df54d8f625a80fcf0231a0/macro/derive/src/lib.rs#L12)

File: macro/README.md [hunk 3]
```diff
@@ -1,66 +1,22 @@
...
-## Description
+This crate contains macros and attributes for Iroha projects:
 
-This crate contains macros and attributes for Iroha projects.
-`iroha_derive` contains derive macros.
-
-### Features
-
-* `log` attribute for debugging functions inputs and output
+- `iroha_derive` is a custom derive macros for Iroha
```
Comment (Arjentix) [inline macro/README.md:5]: Well, not quite right. `iroha_derive` is just another crate (we have to use crate inside crate for this kind of macros, it's a Rust requirement).
I think, you wanted to describe available macros. The only macro available now is `FromVariant`. I've left a link to it in another comment

**Review by Arjentix — CHANGES_REQUESTED** [review]

File: macro/README.md (hunk 2 above)
Comment (outoftardis) [reply macro/README.md]: I've added the example, made a minor change in the comments there

File: macro/README.md (hunk 3 above)
Comment (outoftardis) [reply macro/README.md]: fixed (I think)

**Review by Arjentix — APPROVED** [review]

**Review by appetrosyan — APPROVED** [review]

**Pull request merged**